content/developers/developer-patterns/navigating-merklelogs/index.md

```diff
@@ -0,0 +1,658 @@
+                          \
+                6           13             22        (affectionately called the alpine zone)
+                  \            \             \
+        1 |  2  |  5  |   9  |  12  |  17  | 21  | -- massif 'tree line'
+          |     |     |      |      |      |     |
+        0 |0   1| 3  4| 7   8|10  11|15  16|18 19|
+          |-----|-----|------|------|------|-----|
+          |  0  |  1  |  2   |   3  |   4  |  5  | MASSIF's
+```
+
+The sibling *proof* path for the leaf with `mmrIndex` 7 would be [8, 12, 6], and the "peak bagging" algorithm would then be applied to get the root if that was desired.
+
+```output
+H(7 || 8) = 9
+H(9 || 12) = 13
+H(6 || 13) = 14
+```
+
+14 is a peak in the MMR. The precise set of peaks are determined exclusively from the current mmr size which is 23 in this case. Showing a leaf has an authentication path to a peak for the MMR is sufficient to prove its inclusion in the MMR.
+
+## The "look back" peak stack can be visualized like this
+
+A specific challenge for log implementations with very large data sets is  answering "how far back" or "how far forward" may I need to look?
+
+MMRs differ from classic binary Merkle trees in how the incomplete sub-trees are combined into a common root.
+For an MMR, the common root is defined by an algorithm for combining the adjacent, incomplete sub-trees.
+Rather than by the more traditional, temporary, assignment of un-balanced siblings.
+Such un-balanced siblings would later have to be re-assigned (balanced) when there were sufficient leaves to merge the sub-trees.
+
+While a single root can be produced for an MMR, due to the properties of the accumulator construction, there is much less value in doing so. Typically, authentication use cases are better served by paths to an accumulator.
+
+This detail is what permits us to publish the log data immediately that your events are added to the log.
+
+So the specific properties of Merkle Mountain Ranges lead to an efficiently computable and stable answer to the question of "which other nodes do I need".  Such that we *know* categorically that we do not need to look *forward* of the current massif and further we know precisely which nodes we need from the previous massifs.
+
+The "free nodes" in the alpine zone always require "ancestors" from previous nodes when producing inclusion proofs that pass through them and when adding new nodes to the end of the log. But those ancestors are precisely the peaks forming the accumulator that we want anyway so we can attest to the log state and provide efficient, permanently useful, proofs of inclusion and consistency.
+
+The progression of the peak stack (accumulator) can be visualized like this. In this diagram we gather the dependent nodes into the massifs they belong too and carry forward the peak stack (accumulator) that fully authenticates the entire state of the MMR preceding the massif.
```

Comment:
For the intended audience of this document, how likely is it a reader will choose the (slightly more common) 'verify a user's identity' definition of 'authenticate'?

Reply:
I missed this one when I policed up the other references. Will sort this one on monday